chartmogul/resource.py

```diff
@@ -191,7 +191,7 @@ def _validate_arguments(cls, method, kwargs):
             )
 
     @classmethod
-    def _method(callerClass, method, http_verb, path=None, useCallerClass=False):
+    def _method(callerClass, method, http_verb, path=None, useCallerClass=False, useUUIDFor=None):
```

Comment:
If there is a better way, please feel free to comment on it

Reply:
The `useUUIDFor` being used only for this endpoint seems like a code smell to me. There should be a better way to do this. And perhaps doing what I suggested before (https://github.com/chartmogul/platform/pull/19166#issuecomment-1853150940), regarding supporting both `/customers/:customer_uuid/notes` and `/customer_notes` for both listing all notes and creating a note should help this.

The end result would be something like this:

```python
Customer.notes = CustomerNote._method("all", "get", "/customers/{uuid}/notes", useCallerClass=True)
Customer.createNote = CustomerNote._method("create", "post", "/customers/{uuid}/notes", useCallerClass=True)
```

Just like how I did it for contacts:

https://github.com/chartmogul/chartmogul-python/blob/ed81ce83988c546b1097809643cada6774a4ff24/chartmogul/api/customer.py#L82-L85

This means we have to modify the core API code to support both endpoints as such. What do you think?

Reply:
@SoeunSona The `useUUIDFor` being used only for this endpoint seems like a code smell to me. There should be a better way to do this. And perhaps doing what I suggested before (https://github.com/chartmogul/platform/pull/19166#issuecomment-1853150940), regarding supporting both `/customers/:customer_uuid/notes` and `/customer_notes` for both listing all notes and creating a note should help this.

The end result would be something like this:

```python
Customer.notes = CustomerNote._method("all", "get", "/customers/{uuid}/notes", useCallerClass=True)
Customer.createNote = CustomerNote._method("create", "post", "/customers/{uuid}/notes", useCallerClass=True)
```

Just like how I did it for contacts:

https://github.com/chartmogul/chartmogul-python/blob/ed81ce83988c546b1097809643cada6774a4ff24/chartmogul/api/customer.py#L82-L85

This means we have to modify the core API code to support both endpoints as such. What do you think?

Reply:
@briwa Thank you for suggesting this idea. But It seems overkill because we need to change not only the core api but also change all the libraries. And it also requires communication cost with product team to update. I think it would be better to release without this

Reply:
I agree that this isn't great, and @briwa solution is a good one, but I think this is a compromise we can live with, specially because even if we then change the underlying API we can keep this around for backwards compatibility.